jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    public void refreshCommentsView() {
+        setCurrentPage(1);
+        getTaskCommentView().clearCommentInputForm();
+        loadTaskComments();
+    }
+
+    public void loadMoreTaskComments() {
+        setCurrentPage(getCurrentPage() + 1);
+        loadTaskComments();
+    }
+
+    public void sortComments(final boolean sortAsc) {
+        setSortAsc(sortAsc);
+        refreshCommentsView();
+    }
+
+    public void addTaskComment(final String text) {
+        if (forLog) {
+            return;
+        }
+        taskService.call((Void response) -> {
+                             refreshCommentsView();
+                             getTaskCommentView().resetPagination();
+                         }
+        ).addTaskComment(getServerTemplateId(),
+                         getContainerId(),
+                         getTaskId(),
+                         text,
+                         new Date());
+    }
+
+    public void removeTaskComment(long commentId) {
+        taskService.call((Void response) -> {
+                             refreshCommentsView();
+                             getTaskCommentView().displayNotification(constants.CommentDeleted());
+                         }
+        ).deleteTaskComment(getServerTemplateId(),
+                            getContainerId(),
+                            getTaskId(),
+                            commentId);
+    }
+
+    public void onTaskSelectionEvent(@Observes final TaskSelectionEvent event) {
+        setSelectedTask(event);
+        forAdmin = event.isForAdmin();
+        forLog = event.isForLog();
+        getTaskCommentView().newCommentsHidden(forLog == true);
+        refreshCommentsView();
+    }
+
+    public Predicate<CommentSummary> getDeleteCondition() {
+        return c -> forLog == false && (forAdmin || c.getAddedBy().equals(identity.getIdentifier()));
+    }
+
+    public void onTaskRefreshedEvent(@Observes final TaskRefreshedEvent event) {
+        if (isSameTaskFromEvent().test(event)) {
+            refreshCommentsView();
+        }
+    }
+
+    public void onTaskCompletedEvent(@Observes final TaskCompletedEvent event) {
+        if (isSameTaskFromEvent().test(event) && forLog == false) {
+            forLog = true;
+            getTaskCommentView().newCommentsHidden(true);
+            refreshCommentsView();
+        }
+    }
+
+    @Inject
+    public void setTaskService(final Caller<TaskService> taskService) {
+        this.taskService = taskService;
+    }
+
+    @Inject
+    public void setIdentity(User identity) {
+        this.identity = identity;
+    }
+
+    public interface TaskCommentsView extends UberElement<AbstractTaskCommentsPresenter> {
+
+        void newCommentsHidden(Boolean hidden);
+
+        void clearCommentInputForm();
+
+        void setCommentList(final List<CommentSummary> commentList);
+
+        void resetPagination();
+
+        void hideLoadButton(boolean hidden);
+
+        void displayNotification(String text);
```

Comment:
This is just an event to show the message, I would suggest keeping it in the presenter instead.